internal/api/commitment_test.go

```diff
@@ -1325,3 +1325,109 @@ func Test_ConvertCommitments(t *testing.T) {
+		Body:   assert.JSONObject{"duration": "2 hours"},
+		ExpectBody: assert.JSONObject{"commitment": assert.JSONObject{
+			"id":                1,
+			"service_type":      "second",
+			"resource_name":     "capacity",
+			"availability_zone": "az-one",
+			"amount":            10,
+			"unit":              "B",
+			"duration":          "2 hours",
+			"created_at":        s.Clock.Now().Add(-1 * time.Hour).Unix(),
+			"creator_uuid":      "uuid-for-alice",
+			"creator_name":      "alice@Default",
+			"can_be_deleted":    true,
+			"confirmed_at":      s.Clock.Now().Add(-1 * time.Hour).Unix(),
+			"expires_at":        s.Clock.Now().Add(1 * time.Hour).Unix(),
+		}},
+		ExpectStatus: http.StatusOK,
+	}.Check(t, s.Handler)
+
+	// Positive: Pending commitment
+	assert.HTTPRequest{
+		Method: http.MethodPost,
+		Path:   "/v1/domains/uuid-for-germany/projects/uuid-for-berlin/commitments/new",
+		Body: assert.JSONObject{
+			"commitment": assert.JSONObject{
+				"service_type":      "second",
+				"resource_name":     "capacity",
+				"availability_zone": "az-one",
+				"amount":            10,
+				"confirm_by":        s.Clock.Now().Add(1 * day).Unix(),
+				"duration":          "1 hours",
+			},
+		},
+		ExpectStatus: http.StatusCreated,
+	}.Check(t, s.Handler)
+
+	assert.HTTPRequest{
+		Method: http.MethodPost,
+		Path:   "/v1/domains/uuid-for-germany/projects/uuid-for-berlin/commitments/2/update-duration",
+		Body:   assert.JSONObject{"duration": "2 hours"},
+		ExpectBody: assert.JSONObject{"commitment": assert.JSONObject{
+			"id":                2,
+			"service_type":      "second",
+			"resource_name":     "capacity",
+			"availability_zone": "az-one",
+			"amount":            10,
+			"unit":              "B",
+			"duration":          "2 hours",
+			"created_at":        s.Clock.Now().Unix(),
+			"creator_uuid":      "uuid-for-alice",
+			"creator_name":      "alice@Default",
+			"can_be_deleted":    true,
+			"confirm_by":        s.Clock.Now().Add(1 * day).Unix(),
+			"expires_at":        s.Clock.Now().Add(2*time.Hour + 1*day).Unix(),
+		}},
+		ExpectStatus: http.StatusOK,
+	}.Check(t, s.Handler)
+
+	// Negative: Provided date is invalid
+	assert.HTTPRequest{
+		Method:       http.MethodPost,
+		Path:         "/v1/domains/uuid-for-germany/projects/uuid-for-berlin/commitments/1/update-duration",
+		Body:         assert.JSONObject{"duration": "3 hours"},
+		ExpectBody:   assert.StringData("provided duration: 3 hours does not match the config [1 hour 2 hours]\n"),
+		ExpectStatus: http.StatusUnprocessableEntity,
+	}.Check(t, s.Handler)
+
+	// Negative: Provided Date < Commitment Duration
+	assert.HTTPRequest{
+		Method:       http.MethodPost,
+		Path:         "/v1/domains/uuid-for-germany/projects/uuid-for-berlin/commitments/1/update-duration",
+		Body:         assert.JSONObject{"duration": "1 hour"},
+		ExpectBody:   assert.StringData("provided duration: 1 hour cannot be extended with config [1 hour 2 hours]\n"),
```

Comment:
This error message sounds odd, because it does not explain that we are trying to shorten the duration. I would like to see something like:
```suggestion
		ExpectBody:   assert.StringData("duration change from 2 hours to 1 hour forbidden\n"),
```

Reply:
The response now matches your suggestion.